lib/getWorkspacesYarn.js

```diff
@@ -22,13 +23,21 @@ function getWorkspacesYarn(cwd) {
 		throw new TypeError("package.json: workspaces or workspaces.packages: Must be non-empty array of string");
 	}
 
+	// Ignore packages via CLI or via manifest workspaces
+	const _ignorePackages = ignorePackages
+		? ignorePackages
+		: packages.filter((p) => /^!/.test(p)).map((p) => p.replace("!", ""));
+
+	// Remove ignoredPackages from packages
+	if (_ignorePackages.length > 1) packages = packages.filter((p) => !/^!/.test(p));
```

Comment:
No need for this. Have a look: https://github.com/sindresorhus/multimatch/blob/main/test/test.js
```
t.deepEqual(multimatch(['foo', 'bar', 'baz'], ['foo', '!bar']), ['foo']);
```